objstore/objstore.go

```diff
@@ -0,0 +1,112 @@
+package objstore
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"fx.prodigy9.co/config"
+	"github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+)
+
+type StorageClient struct {
+	accessKey string
+	secretKey string
+	url       string
+	bucket    string
+	client    *minio.Client
+}
+
+var (
+	StorageAccessKeyConfig = config.Str("STORAGE_ACCESS_KEY")
+	StorageSecretKeyConfig = config.Str("STORAGE_SECRET_KEY")
+	StorageURLConfig       = config.Str("STORAGE_URL")
+	StorageBucketConfig    = config.Str("STORAGE_BUCKET_NAME")
+
+	DefaultClient *StorageClient
+)
+
+func NewClient(cfg *config.Source) *StorageClient {
+	return newS3Client(cfg)
+}
+
+func getDefaultClient() {
+	if DefaultClient == nil {
+		DefaultClient = newS3Client(config.Configure())
+	}
+}
+
+func newS3Client(cfg *config.Source) *StorageClient {
+	if cfg == nil {
+		cfg = config.Configure()
+	}
+
+	accessKey := config.Get(cfg, StorageAccessKeyConfig)
+	secretKey := config.Get(cfg, StorageSecretKeyConfig)
+	url := config.Get(cfg, StorageURLConfig)
+	bucket := config.Get(cfg, StorageBucketConfig)
+
+	client, err := minio.New(url, &minio.Options{
+		Creds:  credentials.NewStaticV4(accessKey, secretKey, ""),
+		Secure: true,
+	})
+	if err != nil {
+		log.Println("objstore: ", err)
+	}
+
+	return &StorageClient{
+		accessKey: accessKey,
+		secretKey: secretKey,
```

Comment:
After `minio.New`, do we still need accessKey and secretKey?

Reply:
deleted krub, including the url because it was only used when creating it for the first time.